azure_data_cosmos_engine/src/query/producer/read_many.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::collections::{HashMap, VecDeque};
+
+use crate::{
+    query::{node::PipelineNodeResult, query_result::QueryResultShape, DataRequest, QueryResult},
+    ErrorKind,
+};
+
+use super::{create_query_chunk_states, state::QueryChunkState};
+
+pub struct ReadManyStrategy {
+    pub query_chunks: Vec<HashMap<String, Vec<(usize, String, String)>>>,
+    pub current_query_chunk_index: usize,
+    pub query_chunk_states: Vec<QueryChunkState>,
+    pub items: VecDeque<QueryResult>,
+    pub result_shape: QueryResultShape,
+}
+
+impl ReadManyStrategy {
+    pub fn new(query_chunks: Vec<HashMap<String, Vec<(usize, String, String)>>>) -> Self {
+        let query_chunk_states = create_query_chunk_states(&query_chunks);
+        tracing::debug!("initialized query chunk states: {:?}", query_chunk_states);
+        Self {
+            query_chunks: query_chunks,
+            current_query_chunk_index: 0,
+            query_chunk_states: query_chunk_states,
+            items: VecDeque::new(),
+            result_shape: QueryResultShape::RawPayload,
+        }
+    }
+
+    pub fn requests(&mut self) -> Vec<DataRequest> {
+        // Here is where we would create DataRequests for the current query chunk.
+
+        // In the unordered strategy, we simply return the first partition key range's request.
+        // Once that partition is exhausted, we remove it from the list and return the next one.
+        let mut requests = Vec::new();
+        tracing::debug!(query_chunk_index = ?self.current_query_chunk_index, "getting requests for current query chunk");
+        tracing::debug!(query_chunk_states = ?self.query_chunk_states.get(self.current_query_chunk_index), "current query chunk state");
+        while requests.is_empty() {
+            // If there are no more partitions, return None.
```

Comment:
This doesn't return `None`.

Reply:
Yeah, that _was_ the contract of `requests` before, but I refactored it recently. When there are no requests, just return an empty `Vec`.